File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-1.js
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     constructor(val = 0, next = null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

class Solution {
    /**
     * @param {ListNode} list1
     * @param {ListNode} list2
     * @return {ListNode}
     */
    mergeTwoLists(list1, list2) {
        const res= new ListNode()
        let current=res
        while (list1 && list2 )
        {
            if(list1.val <= list2.val)
            {
                current.next=list1
                list1=list1.next
            }
            else{
                current.next=list2
                list2=list2.next
            }
            current=current.next
        }
        // Append remaining nodes of list1 or list2
        if (list1) {
            current.next = list1;
        } else {
            current.next = list2;
        }
        return res.next
        
    }
}

File: Data Structures & Algorithms/reverse-a-linked-list/submission-1.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */

func reverseList(head *ListNode) *ListNode {
    
	if head==nil {
		return head
	}

	var prev *ListNode
	current:=head

	for current !=nil {
		temp:=current.Next
		current.Next=prev
		prev=current
		current=temp
	}

	head=prev

	return head
}
